python_on_whales/components/network/cli_wrapper.py

```diff
@@ -209,14 +217,30 @@ def disconnect(
         full_cmd += [network, container]
         run(full_cmd)
 
+    def exists(
+        self,
+        network: ValidNetwork,
+    ) -> bool:
+        """Check if a network exists
+
+        Parameters:
+            network: The name of the network.
+        """
+        try:
+            self.inspect(network)
+        except NoSuchNetwork:
+            return False
+        else:
+            return True
+
     @overload
-    def inspect(self, x: str) -> Network: ...
+    def inspect(self, x: ValidNetwork) -> Network: ...
```

Comment:
`inspect()` doesn't make a lot of sense to take a `Network` object, since it's used for *getting* a `Network` object. Can you instead make `exists()` do `self.inspect(str(network))`? (or check how his is handled for other object types such as `Container`, `Pod`, ...)

Reply:
It seems like things are slightly inconsistent on whether or not `inspect()` takes `ValidSomeObject` or just `str`. This change was based on `ContainerCLI.inspect()`, which takes `ValidContainer` as an argument. `PodCLI.inspect()` similarly takes `ValidPod`. However, `VolumeCLI.inspect()` and `ServiceCLI.inspect()` only take a `str`.

Given the inconsistency, sounds like `str` would be the preferred approach, yes?

Reply:
Oh dear 🥲 I don't mind - the main thing I noticed is you didn't quite update the type annotations correctly in adding the support. I'd leave it as just `str` here to keep the scope of the changeset contained.

Reply:
Moved back to `str` and used `self.id` in `Network.exists()`.

You'll have to excuse any errors in my type changes, I'm not all that familiar with the codebase. Been using it a lot the last several months, but only recently dived into the actual code. For my own understanding, what was incorrect with how I'd updated the type annotations?

Reply:
> For my own understanding, what was incorrect with how I'd updated the type annotations?

In [this commit](https://github.com/gabrieldemarmiesse/python-on-whales/commit/8d27a80ba7ade1ec7b9e66346f7cad0b90470d2d) you had
```python
    @overload
    def inspect(self, x: ValidNetwork) -> Network: ...

    @overload
    def inspect(self, x: List[str]) -> List[Network]: ...

    def inspect(self, x: Union[str, List[str]]) -> Union[Network, List[Network]]:
        if isinstance(x, str) or isinstance(x, Network):
            return Network(self.client_config, x)
        else:
            return [Network(self.client_config, reference) for reference in x]
```

* The `@overload` methods say that the two possible invocations are a single `ValidNetwork` (`str | Network`) or a list of `str`. This means there's not support for a list of `Network`, i.e. the second overload should've had `List[ValidNetwork]` (although `List` is a bad choice of type, see https://github.com/gabrieldemarmiesse/python-on-whales/issues/584).
* The actual method signature (without `@overload`) wasn't updated to support `Network` at all, so is inconsistent with the overloads.

If we wanted to support `Network` objects here I would make this:
```python
    @overload
    def inspect(self, x: ValidNetwork) -> Network: ...

    @overload
    def inspect(self, x: Iterable[ValidNetwork]) -> List[Network]: ...

    def inspect(self, x: Union[ValidNetwork, Iterable[ValidNetwork]]) -> Union[Network, List[Network]]:
        if isinstance(x, Iterable) and not isinstance(x, str):
            return [Network(self.client_config, str(net)) for net in x]
        else:
            return Network(self.client_config, str(x))
```

Reply:
To be clear, I still think the `exists()` method should support `Network` types:
```python
    def exists(self, network: ValidNetwork) -> bool:
        try:
            self.inspect(str(network))
        except NoSuchNetwork:
            return False
        else:
            return True
```

Reply:
Got it. Overloads are one of things I haven't touched much with type annotations so it doesn't entirely surprise me that that would be where I went awry. Thanks for the clarification!